File: 백준/Silver/2828. 사과 담기 게임/사과 담기 게임.py
length, basket = map(int, input().split())
apple = int(input())

apple_arr = []
for _ in range(apple):
  apple_arr.append(int(input()))

start, end = 1, basket
count = 0
for item in apple_arr:
  while(True):
    if start <= item and end >= item:
      break
    elif start > item:
      start -= 1
      end -= 1
    elif end < item:
      start += 1
      end += 1
    
    count += 1

print(count)

File: 백준/Silver/2828. 사과 담기 게임/사과 담기 게임.c
#include <stdio.h>

int main(void) { 
   int length, basket, apple;
   scanf("%d %d", &length, &basket);
   scanf("%d", &apple);

   int apple_arr[apple];
   for(int i = 0; i < apple; i++){
      scanf("%d", &apple_arr[i]);
   }

   int start = 1, end = basket, count = 0;
   
   for(int i = 0; i < apple; i++){
      while(1){
         if(apple_arr[i] >= start && apple_arr[i] <= end){
            break;
         }
         else if(apple_arr[i] > end){
            start += 1;
            end += 1;
         }
         else if(apple_arr[i] < start){
            start -= 1;
            end -= 1;
         }
         count += 1;
      }      
   }

   printf("%d", count);
   
   return 0;
}